src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/FileRenameTracker.cs

```diff
@@ -0,0 +1,100 @@
+﻿//--------------------------------------------------------------------------------------------
+// FileRenameTracker
+//
+// Exports an IProjectChangeHintReceiver to listen to file renames. If the file being renamed
+// is a code file, it will prompt the user to rename the class to match. The rename is done
+// using code model
+//
+// Copyright(c) 2015 Microsoft Corporation
+//--------------------------------------------------------------------------------------------
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using Task = System.Threading.Tasks.Task;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    [Export(typeof(IProjectChangeHintReceiver)), Export]
+    [ProjectChangeHintKind(ProjectChangeFileSystemEntityRenameHint.RenamedFileAsString)]
+    [AppliesTo(ProjectCapability.CSharp)]
+    internal class FileRenameTracker : IProjectChangeHintReceiver
+    {
+        private readonly IUnconfiguredProjectVsServices _unconfiguredProjectVsServices;
+        private IComponentModel _componentModel;
+        private VisualStudioWorkspace _visualStudioWorkspace;
+
+        /// <summary>
+        /// The thread handling service.
+        /// </summary>
+        [Import]
+        private IProjectThreadingService ThreadingService { get; set; }
```

Comment:
You can get this from `_unconfiguredProjectVsServices.ThreadingService` instead of importing it.